Cesium3DTilesSelection/src/WebMapTileServiceRasterOverlay.cpp

```diff
@@ -0,0 +1,333 @@
+  }
+
+private:
+  std::string _url;
+  std::vector<IAssetAccessor::THeader> _headers;
+  bool _useKVP;
+  std::string _format;
+  std::string _layer;
+  std::string _style;
+  std::string _tileMatrixSetID;
+  std::optional<std::vector<std::string>> _labels;
+  std::optional<std::map<std::string, std::string>> _staticDimensions;
+  std::vector<std::string> _subdomains;
+};
+
+WebMapTileServiceRasterOverlay::WebMapTileServiceRasterOverlay(
+    const std::string& name,
+    const std::string& url,
+    const std::vector<IAssetAccessor::THeader>& headers,
+    const WebMapTileServiceRasterOverlayOptions& wmtsOptions,
+    const RasterOverlayOptions& overlayOptions)
+    : RasterOverlay(name, overlayOptions),
+      _url(url),
+      _headers(headers),
+      _options(wmtsOptions) {}
+
+WebMapTileServiceRasterOverlay::~WebMapTileServiceRasterOverlay() {}
+
+Future<RasterOverlay::CreateTileProviderResult>
+WebMapTileServiceRasterOverlay::createTileProvider(
+    const CesiumAsync::AsyncSystem& asyncSystem,
+    const std::shared_ptr<CesiumAsync::IAssetAccessor>& pAssetAccessor,
+    const std::shared_ptr<CreditSystem>& pCreditSystem,
+    const std::shared_ptr<IPrepareRendererResources>& pPrepareRendererResources,
+    const std::shared_ptr<spdlog::logger>& pLogger,
+    CesiumUtility::IntrusivePointer<const RasterOverlay> pOwner) const {
+
+  pOwner = pOwner ? pOwner : this;
+
+  const std::optional<Credit> credit =
+      this->_options.credit ? std::make_optional(pCreditSystem->createCredit(
+                                  this->_options.credit.value(),
+                                  pOwner->getOptions().showCreditsOnScreen))
+                            : std::nullopt;
+
+  bool hasError = false;
+  std::string errorMessage;
+
+  std::string layer;
+  std::string style;
+  std::string tileMatrixSetID;
+  if (!_options.layer.has_value()) {
+    hasError = true;
+    errorMessage = "layer is required.";
+  } else {
+    layer = _options.layer.value();
+  }
+  if (!_options.style.has_value()) {
+    hasError = true;
+    errorMessage = "style is required.";
+  } else {
+    style = _options.style.value();
+  }
+  if (!_options.tileMatrixSetID.has_value()) {
+    hasError = true;
+    errorMessage = "tileMatrixSetID is required.";
+  } else {
+    tileMatrixSetID = _options.tileMatrixSetID.value();
+  }
+  std::string format = _options.format.value_or("image/jpeg");
+  uint32_t tileWidth = _options.tileWidth.value_or(256);
+  uint32_t tileHeight = _options.tileHeight.value_or(256);
+
+  uint32_t minimumLevel = std::numeric_limits<uint32_t>::max();
+  uint32_t maximumLevel = 0;
+
+  if (maximumLevel < minimumLevel && maximumLevel == 0) {
+    // Min and max levels unknown, so use defaults.
+    minimumLevel = 0;
+    maximumLevel = 25;
+  }
+
+  bool useKVP;
+
+  std::regex bracket("\\{");
+  std::regex bracketS("\\{s\\}");
+  std::smatch matchBracket;
+  bool hasMatch = std::regex_search(_url, matchBracket, bracket);
```

Comment:
CesiumJS uses regular expressions here because they're "basically free" in the browser context. But `std::regex` is a pretty huge library from what I understand, and it's being used here to do nothing more check if one string contains another. That can be easily done without regular expressions, so let's do that.